From 9674bc06aadffa0533a1a583aa1559ec5d1bb2b9 Mon Sep 17 00:00:00 2001
From: David Mercado <david.mercado@windriver.com>
Date: Tue, 19 Nov 2013 17:54:21 -0500
Subject: [PATCH 138/158] LSI AXM55XX: Add GENERIC_LOCKBREAK

Add GENERIC_LOCKBREAK to arch/arm/Kconfig. The AXM55xx will not boot
without it in 3.10.

Signed-off-by: David Mercado <david.mercado@windriver.com>
---
 arch/arm/Kconfig | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 94f4975..56095f4 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -154,12 +154,17 @@ config TRACE_IRQFLAGS_SUPPORT
 	bool
 	default y
 
+config GENERIC_LOCKBREAK
+	bool
+	default y
+	depends on SMP
+
 config RWSEM_GENERIC_SPINLOCK
 	bool
 	default y
 
 config RWSEM_XCHGADD_ALGORITHM
-	bool
+	def_bool !RWSEM_GENERIC_SPINLOCK && !PREEMPT_RT_FULL
 
 config ARCH_HAS_ILOG2_U32
 	bool
-- 
1.8.1.2

